### PR TITLE
Adding note about kubectl 1.6 to the dev guide

### PR DESCRIPTION
**Pull request by arschles**

### CODE REVIEW

File: docs/DEVGUIDE.md [hunk 1]
```diff
@@ -264,7 +275,7 @@ Next, we'll register a service broker with the catalog.  To do this, we'll
 create a new [`Broker`](../contrib/examples/walkthrough/ups-broker.yaml)
 resource:
 
-    $ kubectl create -f contrib/examples/walkthrough/ups-broker.yaml
+    $ ./kubectl16 create -f contrib/examples/walkthrough/ups-broker.yaml
```
Comment (pmorie) [inline docs/DEVGUIDE.md]: Really, really do not want to change all of these -- how about a note saying the `kubectl` must resolve to a 1.6 kubectl.

**Comment by MHBauer** [comment]
Also fix line 71.

File: docs/DEVGUIDE.md (hunk 1 above)
Comment (arschles) [inline docs/DEVGUIDE.md]: we only need to change the `create` lines. you can use a `kubectl` 1.5.x for all other operations

Comment (pmorie) [inline docs/DEVGUIDE.md]: Do you really feel being on <= 1.6-candidate is that common?

Comment (arschles) [inline docs/DEVGUIDE.md]: yes. I haven't seen docs that tell users to download a 1.6 beta version of `kubectl`

Comment (pmorie) [inline docs/DEVGUIDE.md]: I would be fine saying you need to use 1.6 kubectl. This is bleeding edge alpha. I think making it very clear that this required is sufficient, personally

**Comment by arschles** [comment]
@MHBauer done

**Comment by MHBauer** [comment]
Given that the only thing done before the get is installing the charts with helm, I don't understand why my get would not work with a v1.5 kubectl.

Does helm use kubectl in the background?

**Comment by arschles** [comment]
@MHBauer answers below:

>Given that the only thing done before the get is installing the charts with helm, I don't understand why my get would not work with a v1.5 kubectl.

Get does work with a v1.5 `kubectl`. Only `creates` need 1.6. @pmorie has requested that we use 1.6 for all operations to keep things simpler for the user

>Does helm use kubectl in the background?

No. It sends rendered templates over gRPC to Tiller, which uses the Go SDK to send operations to the Kubernetes core server.

**Comment by MHBauer** [comment]
Totally fine with requiring v1.6 kubectl as client. Still attempting the walkthrough.

**Comment by arschles** [comment]
@MHBauer cool. Note that this PR isn't intended to fix the walkthrough completely. https://github.com/kubernetes-incubator/service-catalog/pull/593 is, intended to do so.

**Comment by jpeeler** [comment]
Assuming consensus to simply require kubectl 1.6 or greater, the oldest commit needs updating to be consistent with that decision.

**Pull request merged**